### ISSUE TITLE
[no-unused-vars] new v4 false positive: generic type arg used as function parameter list

### ISSUE THREAD

**Issue by mmkal** [typescript-eslint/typescript-eslint#2449]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
      "warn",
      { "argsIgnorePattern": "^_", "ignoreRestSiblings": true }
    ]
  }
}
```

```typescript
export type F<A extends unknown[]> = (...a: A) => unknown;
```

**Expected Result**

No error

**Actual Result**

Error - `'A' is defined but never used.eslint@typescript-eslint/no-unused-vars`

**Additional Info**

When removing the `...` before `a`, the error is no longer reported (although obviously this is a completely different type - just including this info in case it's helpful). i.e. this is fine:

```typescript
export type F<A extends unknown[]> = (a: A) => unknown;
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `12.17.0` |

**Comment by backbone87** [comment]
Just an addition:
```
import { Foo } from './bar';

export type F<A extends unknown[]> = (...a: Foo<A>) => unknown;
```
In this example A as well as Foo are reported as unused.
